apps/video/js/video.js

```diff
@@ -114,6 +115,17 @@ document.addEventListener('visibilitychange', function visibilityChange() {
     }
   }
   else {
+    // Bug 1151775
+    // If the app is coming to the foreground before a sharing activity is
+    // complete, restore the video (the video will have been released before
+    // initiating the sharing activity and is only restored when the sharing
+    // activity is completed). When the sharing activity is complete and
+    // restoreVideo is again called, the restoreVideo function takes care of
+    // not restoring the video when it doesn't need to be restored.
+    if (sharingActivity) {
+      videoHardwareReleased = true; // Otherwise restoreVideo will abort
+      restoreVideo();
+    }
```

Comment:
I'm not sure why I concluded there needed to be a new variable. As you can see by line 126, I has assumed 'videoHardwareReleased' was not true in the STR. I don't have an explanation as to why I thought that.
Regarding "disposition=window" activities potentially not completing, do you mean where the initiator will not ever get an onsuccess or onerror callback? I'm finding that killing the email app in the middle of the activity results in the 'onerror' callback being called. For my information, can you explain under what circumstances the activity will not complete?